pkg/btp/Authentication.go

```diff
@@ -0,0 +1,96 @@
+package btp
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+type BTPUtils struct {
+	Exec     ExecRunner
+	loggedIn bool
+}
+
+type LoginOptions struct {
+	Url       string
+	Subdomain string
+	User      string
+	Password  string
+	Tenant    string
+}
+
+func (btp *BTPUtils) LoginCheck() (bool, error) {
+	return btp.loggedIn, nil
```

Comment:
The LoginCheck() method returns an attribute of the BTPUtils struct. It does not check the client is really logged in. I'm not sure about the behaviour of the btp cli, but is it guaranteed that the session is valid for a long period?

E.g. when a clone is running for multiple hours, is the session afterwards still active? The LoginCheck() may return true, but is this correct?

Reply:
Anyway, is there a real benefit for checking the login status?